### PR TITLE
feat(governance): epoch total rewards tables

### LINKED ISSUE #1230: Show rewards and fees earned (no wallet connected) - no pagination
Dependent on summary ticket on vega core that needs to be created @dexturr 

## Story 

As a user who has not connected their wallet
I want to see a total of rewards and fees paid out
So that I can understand the financial benefits of using Vega

## Acceptance criteria 

- [x] I can see rewards and infrastructure fees paid out per epoch, per asset, per type (staking rewards, infrastructure fees, price taking rewards, price making rewards, liquidity provision rewards, market creation rewards)
- [x] I can see a total per asset 
- [x] I can see a description of each type to help me understand where they've come from on mouseover
- [x] There is a link to docs to "[Understand how rewards are calculated ↗](https://docs.vega.xyz/docs/testnet/concepts/trading-on-vega/fees-rewards#trading-rewards)"
- [x] I can see whether there is a delay on reward payout i.e. "Earn rewards and fees for trading and maintaining the network. Rewards are paid out at the end of an epoch." becomes "Earn rewards and fees for trading and maintaining the network. Rewards will be paid out from the treasury [6h] after the epoch ends." when there is a delay to payout
- [x] Epoch list should be shown in descending order with the most recent at the top

## Background / additional details 

- Staking rewards = types.AccountTypeGlobalReward
- Infrastructure fees = types.AccountTypeFeesInfrastructure
- Price taking rewards = types.AccountTypeMakerPaidFeeReward
- Price making rewards = types.AccountTypeMakerReceivedFeeReward
- Liquidity provision rewards = types.AccountTypeLPFeeReward
- Market Creation Rewards = types.AccountTypeMarketProposerReward

Mouseovers per type
 
* ( i ) Staking rewards supplement infrastructure fees in the early stages of the network, rewarding validators and those who stake them for maintaining the network. 
* ( i ) Infrastructure fees are incurred across the network during trading.  They are distributed to validators according to their share of total stake on the network, and passed onto those who stake them, proportionate to their own stake after validator commission is taken.  
* ( i ) Price taking rewards are based on the proportion of the total maker fees you paid while trading, on markets where there is a funded reward
* ( i ) Price making rewards are based on the proportion of the total maker fees you received while trading, on markets where there is a funded reward
* ( i ) Liquidity provision rewards are distributed based on how much you have earned in liquidity fees, funded by a liquidity reward pool for that market.
* ( i ) Market creation rewards are paid out to the creator of any market that exceeds a set threshold of cumulative volume in a given epoch, currently [rewards.marketCreationQuantumMultiple].

## Tasks 

- [x] Work it out 
- [x] Build 

## Sketch

Image

### PR DESCRIPTION
**Pull request by sam-keen**
# Related issues 🔗

Closes #1230

# Description ℹ️

Includes a toggle for displaying totals per epoch (this work) or individual rewards (currently using the old work, to be updated in a separate ticket soon). Currently only using pagination to limit to the most recent 10 so stagnet3 doesn't implode from the weight of them. Proper pagination to follow.

# Demo 📺

Screenshot 2023-02-09 at 13 52 25

### CODE REVIEW

File: apps/token/src/routes/rewards/epoch-total-rewards/epoch-rewards.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const { data, loading, error, refetch } = useEpochAssetsRewardsQuery({
+    variables: {
+      epochRewardSummariesPagination: {
+        first: 10,
+      },
+    },
+  });
+  useRefreshAfterEpoch(data?.epoch.timestamps.expiry, refetch);
+
+  useEffect(() => {
```
Comment (asiaznik) [inline apps/token/src/routes/rewards/epoch-total-rewards/epoch-rewards.tsx:24]: suggestion (non-blocking): I don't think you need the combination of `useEffect` and `useState` here.

Simple `const epochRewardSummaries = generateEpochTotalRewardsList(data) || [];` would do the same job here

File: apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+const displayReward = (reward: string) => {
+  if (Number(reward) === 0) {
+    return <span className="text-vega-dark-300">0</span>;
+  }
+
+  if (reward.split('.')[1] && reward.split('.')[1].length > 4) {
+    return (
+      <Tooltip description={formatNumber(reward)}>
+        <button>{formatNumber(Number(reward).toFixed(4))}</button>
```
Comment (asiaznik) [inline apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx:33]: question (non-blocking): What's the button for?

File: apps/token/src/routes/rewards/epoch-total-rewards/epoch-rewards.tsx (hunk 1 above)
Comment (sam-keen) [reply apps/token/src/routes/rewards/epoch-total-rewards/epoch-rewards.tsx:24]: Great point! Thanks

File: apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx (hunk 2 above)
Comment (sam-keen) [reply apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx:33]: I'd grabbed its example use from ui-toolkit where a button was used, but just tried it without and it worked fine. I'll remove it from here and from the stories in ui-toolkit

File: apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    <div className="h-full w-5 absolute left-0 top-0 bg-gradient-to-l from-transparent to-black pointer-events-none" />
+  </div>
+);
+
+export const EpochTotalRewardsTable = ({
+  data,
+}: EpochTotalRewardsGridProps) => {
+  const { t } = useTranslation();
+
+  const rowData = data.assetRewards.map(({ name, rewards, totalAmount }) => ({
```
[the remark this answers is not in the document]
Comment (edd) [reply apps/token/src/routes/rewards/epoch-total-rewards/epoch-total-rewards-table.tsx:89]: I still find this set of maps/filters hard to navigate even with the added tests - but now I feel a bit more confident that if it's an issue it's going to be easier to figure out where in this we're going wrong, and to safely iterate on it

**Review by edd — APPROVED** [review]
Feedback addressed (in DMS, because it turns out I didn't actually hit submit on my comment)

**Review by asiaznik — APPROVED** [review]

**Review by pscott31 — APPROVED** [review]

**Pull request merged**